tests/unit/test-cmd/test.build.js

```diff
@@ -37,6 +43,112 @@ describe('build', () => {
     );
   });
 
+  it('gives the correct name to a localized extension', () => {
+    return withTempDir(
+      (tmpDir) =>
+        build({
+          sourceDir: fixturePath('minimal-localizable-web-ext'),
+          artifactsDir: tmpDir.path(),
+        })
+        .then((buildResult) => {
+          assert.match(buildResult.extensionPath,
+                       /name_of_the_extension-1\.0\.zip$/);
+          return buildResult.extensionPath;
+        })
+    );
+  });
+
+  it('handles repeating localization keys', () => {
+    return withTempDir(
+      (tmpDir) => {
+        const messageFileName = path.join(tmpDir.path(), 'messages.json');
+        //This is missing the 'message' key
+        fs.writeFileSync(messageFileName,
+          `{"extensionName": {
+              "message": "example extension ",
```

Comment:
heh, you are sneaky ;) but I think it would make more sense to put the space here instead: `'__MSG_extensionName__ __MSG_extensionName__'`